src/dlfcn.c

```diff
@@ -575,6 +575,11 @@ static void *get_address_from_import_address_table( void *iat, DWORD iat_size, v
     if( !is_valid_address( ptr ) ||  ptr < (BYTE *)iat || ptr > (BYTE *)iat + iat_size )
         return NULL;
 
+    /* limit checks to mingw for now as msvc seems to have a different binary layout */
+#ifdef __MINGW32__
+    if( ptr < (BYTE *)iat || ptr > (BYTE *)iat + iat_size )

```

Comment:
Check for `ptr < (BYTE *)iat || ptr > (BYTE *)iat + iat_size` is there two times. Once above `#ifdef __MINGW32__` and once inside of the `#ifdef __MINGW32__`.